Sources/SwiftGenKit/Parsers/Resources/ResourceParser.swift

```diff
@@ -0,0 +1,40 @@
+//
+// SwiftGenKit
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+import PathKit
+
+public enum Resource {
+  public enum ParserError: Error, CustomStringConvertible {
+    case invalidFile(path: Path, reason: String)
+
+    public var description: String {
+      switch self {
+      case .invalidFile(let path, let reason):
+        return "Unable to parse file at \(path). \(reason)"
+      }
+    }
+  }
+
+  // MARK: Resource File Parser
+
+  public class Parser: SwiftGenKit.Parser {
+    private let options: ParserOptionValues
+    var files: [File] = []
+    public var warningHandler: Parser.MessageHandler?
+
+    public required init(options: [String: Any] = [:], warningHandler: Parser.MessageHandler? = nil) throws {
+      self.options = try ParserOptionValues(options: options, available: Parser.allOptions)
+      self.warningHandler = warningHandler
+    }
+
+    public static var defaultFilter: String = ""
+
+    public func parse(path: Path, relativeTo parent: Path) throws {
+      files.append(try File(path: path, relativeTo: parent))
```

Comment:
I think ultimately we want it to be recursive, to list files inside folders deep inside the parsed root path, right?

Reply:
Agreed